lib/shopify_app/session/shop_session_storage.rb

```diff
@@ -26,15 +28,26 @@ def retrieve_by_shopify_domain(domain)
         construct_session(shop)
       end
 
+      def retrieve_access_scopes_by_shopify_domain(domain)
+        shop = find_by(shopify_domain: domain)
+        shop.access_scopes
+      end
+
       private
 
       def construct_session(shop)
         return unless shop
+        begin
+          scopes = shop.access_scopes
+        rescue NotImplementedError, NoMethodError
+          scopes = nil
+        end
```

Comment:
I forget what our original contention was around doing this. Is it still feasible to move this to its own method?

```suggestion
  def scopes(shop)
    begin
      shop.access_scopes
    rescue NotImplementedError, NoMethodError
      nil
    end
  end
```

Reply:
Sure! I see no issues with this!